crates/ide-completion/src/completions.rs

```diff
@@ -50,9 +50,10 @@ use crate::{
 };
 
 /// Represents an in-progress set of completions being built.
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct Completions {
     buf: Vec<CompletionItem>,
+    hide_deprecated: bool,
```

Comment:
I feel like we should do the filtering in the `to_proto` layer, `CompletionItem` retains whether its deprecated so the caller can do the filtering when needed. (that would also be less invasive)

Reply:
That has no way to test...

Reply:
Done that.